omisego/elixir-omg
# feature: run releases barebone

HELLO
## Overview

This should allow you to run child chain and watcher baremetal.

## Changes


- makefile ftw
## Testing
 lemme know if it works

so you want to try baremetal ehm?
First, I suggest you stop any containers you might be running (check if there are any with `docker ps`) and stop `plasma-deployer`, `geth`, `watcher`, `childchain`and `postgres`. Or just do 
1. `docker stop $(docker ps -q)`.
We then go all in.
Run:
2. `make start-services` this starts geth, postgres and plasma-deployer
Beware, plasma-deployer can be flaky. 
3. Open another terminal window an run
`make start-child_chain`
4. Open another terminal window an run
`make start-watcher`
5. Run `make get-alarms` once 3. and 4. are done booting, you should see curl responses from both services.
6. Explore the makefile with `make list`,
7. Other baremetal commands
```
make stop-child_chain
make stop-watcher
make remote-child_chain (allows you to attach yourself to the running node)
make remote-watcher (allows you to attach yourself to the running node)
```
To push code changes:
```
make update-child_chain
make update-watcher
```

Watcher and Child Chain need exported environment variables, they're found bin `bin/variables`.

Thanks @T-Dnzt !
